### ISSUE TITLE
Uncaught rejection for "where"

### ISSUE THREAD

**Issue by zzph** [Vincit/objection.js#2053]
Hiya,

Wonder if this is a bug or intended behaviour.  Unless I mis-understood something- I believe this error isn't being 'caught'.

Reproduction is simple.

```
Person.query().where('pid', '=', undefined).then().catch(err=>console.log('Never gets called'))
// the console.log in catch never gets called
```

```
UnhandledPromiseRejectionWarning:   Error: undefined passed as ... all skipUndefined() ...
```

Is this a bug or am I handling it wrong? 

Ta

**Comment by koskimas** [comment]
The error is thrown when you call `where`, before the query is executed and turned into a promise. In other words, the error is synchronous.

```js
// This will already throw.
Person.query().where('pid', '=', undefined)
```

So `.then(...)` never gets called and the query never gets executed.

I'd always use async/await if possible to avoid this kind of situations

```js
try {
  await Person.query().where('pid', '=', undefined)
} catch (error) {
  // Now we get here if the error is synchronous OR asynchronous.
}
```

**Comment by zzph** [comment]
Thanks. So how can I know when methods return a promise vs are synchronous?  Or is this the case for all queries?

**Comment by koskimas** [comment]
Only `then`, `catch` and `execute` methods execute the query and return a promise. If say, `where` returned a promise, how could you chain another `where` call to it? Or `select` call? Promises don't have objection's query builder methods.

When you call

```js
await Person.query()
```

`await` invokes the `then` method under the hood and executes the query.

**Comment by zzph** [comment]
Ok cool, 

In case it helps other- I've used `express-promise-router` to handle this. Now all my `Objection` errors automatically bubble all my objection errors thought your example error handler.

An aside- I posed on an issue, not waiting to raise an issue unnecessarily. Just noticed it's a couple years old (https://github.com/Vincit/objection.js/issues/273). Lmk if there's a better channel for those kinds of questions. 

👍
